service/src/main/resources/db/migration/V281__message_thread_folder.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE message_thread_folder
+(
+    id                      uuid PRIMARY KEY         NOT NULL DEFAULT ext.uuid_generate_v1mc(),
+    created                 timestamp with time zone NOT NULL DEFAULT now(),
+    updated                 timestamp with time zone NOT NULL DEFAULT now(),
+    name                    text                     NOT NULL DEFAULT '',
+    owner_id                uuid                     NOT NULL REFERENCES message_account (id) ON DELETE CASCADE,
+
+    CONSTRAINT uniq$thread_folder UNIQUE (name, owner_id)
+);
+
+CREATE INDEX idx$thread_folder_owner ON message_thread_folder (id, owner_id);
+CREATE INDEX idx$thread_folder_name ON message_thread_folder (owner_id, name);
```

Comment:
If you change folder matching to be based on id instead of name, you should remove this index and change the unique constraint above to have the column order as `(owner_id, name)`. The unique constraint is backed by an index, and that index would facilitate queries by `owner_id`, and you wouldn't need a separate index for that purpose.